.ci

```diff
@@ -1,133 +1,145 @@
+#!/usr/bin/env groovy
+
+// https://github.com/apartmentlist/ci-shared-library
+@Library("ci-shared-library")_
+
 properties([
   buildDiscarder(
     logRotator(
-      artifactDaysToKeepStr: '',
-      artifactNumToKeepStr: '',
-      daysToKeepStr: '30',
-      numToKeepStr: '100'
+      artifactDaysToKeepStr: "",
+      artifactNumToKeepStr: "",
+      daysToKeepStr: "30",
+      numToKeepStr: "10"
     )
   )
 ])
 
 def k8s_label = "${UUID.randomUUID().toString()}"
 
 pipeline {
-
   environment {
-    APP_NAME = 'pubsub_client'
-    CLOUDSDK_CORE_DISABLE_PROMPTS = '1'
-    GIT_COMMIT_SHORT = sh(script: "printf \$(git rev-parse --short ${GIT_COMMIT})", returnStdout: true)
-    APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE = "true"
+    APP_NAME = "pubsub_client"
+    CI = "true"
+    GIT_COMMIT_SHORT = sh(script: "printf \$(git rev-parse --short $GIT_COMMIT)", returnStdout: true)
+    GIT_MESSAGE = sh(script: "git log --format=%B -n 1 $GIT_COMMIT", returnStdout: true).trim()
+    GIT_USER = sh(script: "git log -1 --pretty=format:'%ae'", returnStdout: true).trim()
+    GITHUB_URL = "https://github.com"
     LANG = "en_US.UTF-8"
     LANGUAGE = "en_US:en"
     LC_ALL = "en_US.UTF-8"
-  }
+    SLACK_CHANNEL = "#releases"
+  } // environment
 
   agent {
     kubernetes {
       label k8s_label
-      defaultContainer 'jnlp'
+      defaultContainer "jnlp"
       yaml """
 apiVersion: v1
 kind: Pod
 spec:
   containers:
-  - name: gcc
-    image: gcc:9.3
-    command:
-    - "tail"
-    - "-f"
-    - "/dev/null"
-    imagePullPolicy: Always
-    resources:
-      requests:
-        memory: "512Mi"
-        cpu: "500m"
-      limits:
-      requests:
-        memory: "512Mi"
-        cpu: "500m"
-  - name: ruby25
-    image: gcr.io/al-development/ruby:2.5.8
-    imagePullPolicy: Always
-    resources:
-      requests:
-        memory: "2048Mi"
-        cpu: "2"
-      requests:
-        memory: "2048Mi"
-        cpu: "2"
-    command:
-    - "tail"
-    - "-f"
-    - "/dev/null"
-  - name: ruby26
-    image: gcr.io/al-development/ruby:2.6.6
+  - name: ruby27
+    image: gcr.io/alist-development/ruby:2.7.5
```

Comment:
This is not a blocker, but the latest `2.7` is `2.7.6`. I don't even know if we have own image, but just pointing out.

Reply:
We don't have `2.7.6` or `3.0.4` yet, but I can make a ticket for that. Trying to push out a release for the Drive team and then I can circle back to adding those newer images.